olcut-config-json/src/main/java/com/oracle/labs/mlrg/olcut/config/json/JsonProvenanceMarshaller.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates.
+ *
+ * Licensed under the 2-clause BSD license.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.oracle.labs.mlrg.olcut.config.json;
```

Comment:
Should this class actually be in com.oracle.labs.mlrg.olcut.provenance.io instead of in config.json?

Reply:
No because then we'd split that package across modules. It could be in `com.oracle.labs.mlrg.olcut.provenance.io.json`, but I should have put the other provenance stuff in there and I didn't. I made the automatic module name to be a prefix of the packages and it wouldn't be if I split it up. So for the moment it's like this and if we make breaking changes in OLCUT 6 we can arrange things more sensibly.

Reply:
Ah - good point.  Well shame on me if I reviewed that and didn't comment on the provenance stuff being under olcut.config! :-)

Reply:
Nope, I think that poor structure is on me. The module name should be `com.oracle.labs.mlrg.olcut.json` and the packages rearranged everywhere accordingly.